Framework/Kernel/src/TimeSeriesProperty.cpp

```diff
@@ -176,6 +176,9 @@ operator==(const TimeSeriesProperty<TYPE> &right) const {
   { // so vectors can go out of scope
     std::vector<DateAndTime> lhsTimes = this->timesAsVector();
     std::vector<DateAndTime> rhsTimes = right.timesAsVector();
+    if (lhsTimes.size() != rhsTimes.size()) {
```

Comment:
:+1: maybe extend the `if (this->m_size != right.m_size) ` above with an ` || this->realSize() != right.realSize()`, then the check will be applied for both below